segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -268,14 +268,11 @@ public QueueStats getQueueStatistics() {
     private void processWritesSync() {
         if (getWriteLedger().ledger.isClosed()) {
             // Current ledger is closed. Execute the rollover processor to safely create a new ledger. This will reinvoke
-            // the write processor upon finish, so
+            // the write processor upon finish, so the writes can be reattempted.
             this.rolloverProcessor.runAsync();
-        } else if (!processPendingWrites()) {
+        } else if (!processPendingWrites() && !this.closed.get()) {
             // We were not able to complete execution of all writes. Try again.
             this.writeProcessor.runAsync();
-        } else {
```

Comment:
This comment in the description seems to contradict this change:

```
The rolloverProcessor is invoked after each iteration of the writeProcessor
```

The rolloverProcessor is invoked after each iteration of the writeProcessor. By removing these lines, we don't guarantee it runs on every iteration, only ledger is closed as in line 269. Could you clarify, please?